src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java

```diff
@@ -0,0 +1,172 @@
+package io.kestra.plugin.notifications.discord;
+
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.tasks.VoidOutput;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.serializers.JacksonMapper;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import lombok.experimental.SuperBuilder;
+import org.apache.commons.io.Charsets;
+import org.apache.commons.io.IOUtils;
+
+import java.awt.*;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+public abstract class DiscordTemplate extends DiscordIncomingWebhook {
+
+    @Schema(
+        title = "Template to use",
+        hidden = true
+    )
+    @PluginProperty(dynamic = true)
+    protected String templateUri;
+
+    @Schema(
+        title = "Map of variables to use for the message template"
+    )
+    @PluginProperty(dynamic = true)
+    protected Map<String, Object> templateRenderMap;
+
+    @Schema(
+        title = "Webhook username who sends"
+    )
+    @PluginProperty(dynamic = true)
+    protected String username;
+
+    @Schema(
+        title = "Webhook profile photo who sends"
+    )
+    @PluginProperty(dynamic = true)
+    protected String photoUrl;
+
+    @Schema(
+        title = "Title"
+    )
+    @PluginProperty(dynamic = true)
+    protected String title;
+
+    @Schema(
+        title = "Embed message"
+    )
+    @PluginProperty(dynamic = true)
+    protected String description;
+
+    @Schema(
+        title = "Thumbnail url"
+    )
+    @PluginProperty(dynamic = true)
+    protected String thumbnail;
+
+    @Schema(
+        title = "Get photo from photoUrl and gets link from websiteUrl"
+    )
+    @PluginProperty(dynamic = true)
+    protected String authorName;
+
+    @Schema(
+        title = "Out of embed message"
+    )
+    @PluginProperty(dynamic = true)
+    protected String content;
+
+    @Schema(
+        title = "Footer text"
+    )
+    @PluginProperty(dynamic = true)
+    protected String footer;
+
+    @Schema(
+        title = "Color of text",
+        description = "Possibilities (ORANGE, RED, BLACK, GREEN, YELLOW, CYAN, WHITE and BLUE)"
+    )
+    @PluginProperty(dynamic = true)
```

Comment:
```suggestion
    @PluginProperty
```
This is not rendered